### DISCUSSION TITLE
Mitochondrial DNA is longer than expected

### DISCUSSION THREAD

**Discussion by Phismil** [Kinggerm/GetOrganelle#269] [Q&A]
Dear Jianjun Jin
Thank you for maintaining the repository.
I am assembling the mitogenome from a wild cat species, and the size  is expected to be ~16K.
However, my assemblies are constantly more than 24K. I tried different Kmers.
Is there any way to define an upper limit for the assembly?
I will appreciate any advise in advance

**Comment by JianjunJin** [comment]
- Please always provide a complete log file.
- For your question, a slimmed assembly graph (*.extend*.fastg in the output directory) is necessary for troubleshooting.

**Reply by Phismil** [reply]
Dear Jianjun Jin
Thank you for your kind response.
Please find attached the requested files.
Regards

[extended_K105.assembly_graph.fastg.extend-animal_mt.fastg.txt](https://github.com/Kinggerm/GetOrganelle/files/11582200/extended_K105.assembly_graph.fastg.extend-animal_mt.fastg.txt)
[get_org.log.txt](https://github.com/Kinggerm/GetOrganelle/files/11582199/get_org.log.txt)

**Reply by JianjunJin** [reply]
Hi, I deeply apologize for my slip. I must have checked it but forgot to respond. 

This is an interesting sample exhibiting "heteroplasmy" in the assembly graph. The depths of the two variants are so even that none of them can be treated as contamination by GetOrganelle, which is why a 24k-chimeric result was generated.

It's difficult to explain why this happens. It may be NUMT or contamination. We observe similar "heteroplasmy" in the graph in our benchmark test (of big cats too!! see [here](https://github.com/Kinggerm/GetOrganelleComparison/blob/master/res/GetOrganelle-animal-mitogenome/auto--SRR3041429-mitogenome/filtered_K87.assembly_graph.fastg.extend-animal_mt.fastg.gz)). You probably know better than me. Looking forward to your opinion on your dataset.

Good luck!

**Comment by imogen-foote** [comment]
Hi, did this issue get solved? I am having a similar problem with get_organelle_from_assembly.py where my mitochondrial genome is coming out around 38Kb but I am expecting ~19K. 
I'm sorry I don't seem to be able to attach the .gfa file format of the slimmed assembly graph, but if you're able to tell me how I can add that too. 
[get_org.log.txt](https://github.com/Kinggerm/GetOrganelle/files/13782128/get_org.log.txt)

**Reply by JianjunJin** [reply]
Hi, thanks for reaching out.

Yes, just change the postfix of the file to `.txt` (like Phismil did), and then you can attach it. Sorry that this is an issue type that I cannot help without the graph file. So please upload. Thanks!

**Reply by imogen-foote** [reply]
Thanks @JianjunJin see attached 
[animal_mt.complete.graph1.selected_graph.gfa.txt](https://github.com/Kinggerm/GetOrganelle/files/13782275/animal_mt.complete.graph1.selected_graph.gfa.txt)

**Reply by imogen-foote** [reply]
Oops I just realised that's the wrong file but just tried to upload the slimmed_assembly_graph.gfa and the file is too large to upload. I don't have the .extend.fastg file output as you mentioned above. See below for the output files I have. Which of these would be helpful?
![image](https://github.com/Kinggerm/GetOrganelle/assets/86809302/6ff1194b-5843-4ea5-a188-710bbd9be733)

**Reply by JianjunJin** [reply]
I guess edge_2239 is a NUMT contig; you may try to annotate it to figure it out. There are other contigs that have animal_mt hits too, which generally has coverages higher than 49.0x. You can run `get_organelle_from_assembly.py` with `--min-depth 48` so that you can have a much smaller slimmed_assembly_graph.gfa to upload. Let me know if you have trouble with it.

**Reply by imogen-foote** [reply]
Thanks @JianjunJin. I tried that (below code) but it's given exactly the same output as before. 
```
get_organelle_from_assembly.py  -F animal_mt \
                                -g $read_dir/assembly_graph.gfa \
                                -o $output_dir \
                                -t 10 \
                                --overwrite \
                                --verbose \
                                --min-depth 48
```
I'm having trouble getting my annotation pipeline to work as well so can't check that just now. Is there anything else I could try?

**Reply by JianjunJin** [reply]
Could you visualize the big graph with Bandage and the `csv` file loaded?

**Reply by imogen-foote** [reply]
Yes, see below. Is there anything in particular that I should be looking for?
![image](https://github.com/Kinggerm/GetOrganelle/assets/86809302/561ae73d-e64b-44ae-a0d6-cc6709f6dcf8)

**Reply by JianjunJin** [reply]
Thanks. I see.
The 38Kb-single contig seems to include the real mt and non-target sequence too. So it is likely to be the disassembly of flye. Redoing the flye assembly may be an option. BTW, what type of data did you use?

**Reply by imogen-foote** [reply]
Aaah ok. I did the flye assembly with ONT data. I tried the assembly with different quality filtering thresholds (was supposed to be Q20+ data but on average came out a little lower than that) and all of them look odd for the mitogenome but the nuclear looks good. I also have some illumina data so I might try the get_organelle_from_reads.py with that and see if that produces a better mitogenome. 
Thanks so much for you help with this issue and for the tool!
